### PR TITLE
syncer/: fix flush checkpoint for ddls for situations that IsSharding is false(#587)

### PR DESCRIPTION
**Pull request by lichunzhu**
cherry-pick #587 

---

### What problem does this PR solve? 
https://github.com/pingcap/dm/issues/514
* When `IsSharding` is false, table checkpoint which contains schema tracker info was not updated in `dm_meta` instantly. But global checkpoint is updated. If worker fails now, it will cause a mismatch between tracker and global checkpoint.
* Fix a dead lock bug.

### What is changed and how it works?
Update table point when we update global checkpoint.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by lichunzhu** [comment]
@csuzhangxc @WangXiangUSTC PTAL

File: syncer/syncer.go [hunk 1]
```diff
@@ -1741,7 +1759,11 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 			needHandleDDLs = appliedSQLs // maybe nil
 		}
 
-		job := newDDLJob(nil, needHandleDDLs, *ec.lastPos, *ec.currentPos, nil, nil, *ec.traceID)
+		if err := s.flushJobs(); err != nil {
```
Comment (csuzhangxc) [inline syncer/syncer.go:1762]: Do we need to `flushJobs()` before `TrySync` as we did in https://github.com/pingcap/dm/pull/625?

File: syncer/syncer.go [hunk 2]
```diff
@@ -1741,7 +1759,11 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
...
+			return err
+		}
+
+		job := newDDLJob(nil, needHandleDDLs, *ec.lastPos, *ec.currentPos, nil, nil, *ec.traceID, sourceTbls)
```
Comment (csuzhangxc) [inline syncer/syncer.go:1766]: Is there any problem to use `sourceTbls` here? see <https://github.com/pingcap/tidb-enterprise-tools/pull/471> 😂.

File: syncer/syncer.go [hunk 3]
```diff
@@ -1712,6 +1727,9 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 		}
 
 		needHandleDDLs = append(needHandleDDLs, sqlDDL)
+		// TODO: current table checkpoints will be deleted in track ddls, but created and updated in flush checkpoints,
```
Comment (csuzhangxc) [inline syncer/syncer.go:1730]: please fix this comment.

**Comment by csuzhangxc** [comment]
@WangXiangUSTC PTAL

**Pull request merged**